deps/rabbitmq_stomp/src/rabbit_stomp_processor.erl

```diff
@@ -1217,3 +1513,504 @@ maybe_apply_default_topic_exchange(Exchange, _DefaultTopicExchange) ->
+to_url(Lol) -> "/" ++ string:join(Lol, "/").
+
+atomise(Name) when is_list(Name) ->
+    list_to_atom(re:replace(Name, "-", "_", [{return,list}, global])).
+
+unescape(Str) -> unescape(Str, []).
+
+unescape("%2F" ++ Str, Acc) -> unescape(Str, [$/ | Acc]);
+unescape([C | Str],    Acc) -> unescape(Str, [C | Acc]);
+unescape([],           Acc) -> lists:reverse(Acc).
+
+
+consume_queue(QRes, Spec0, State = #proc_state{user = #user{username = Username} = User,
+                                                   authz_context = AuthzCtx,
+                                                   queue_states  = QStates0})->
+    check_resource_access(User, QRes, read, AuthzCtx),
+    Spec = Spec0#{channel_pid => self(),
+                  limiter_pid => none,
+                  limiter_active => false,
+                  ok_msg => undefined,
+                  acting_user => Username},
+    rabbit_amqqueue:with_or_die(
+      QRes,
+      fun(Q1) ->
+              case rabbit_queue_type:consume(Q1, Spec, QStates0) of
+                  {ok, QStates} ->
+                      %% io:format("Consume QStates ~p ~n", [QStates]),
+                      %% rabbit_global_counters:consumer_created('STOMP'),
+                      State1 = State#proc_state{queue_states = QStates},
+                              {ok, State1};
+                  {error, Reason} ->
+                      error("Failed to consume from ~s: ~p",
+                                    [rabbit_misc:rs(QRes), Reason],
+                            State)
+              end
+      end).
+
+create_queue(Amqqueue, _State = #proc_state{authz_context = AuthzCtx,
+                                            user = User,
+                                            vhost = VHost}) ->
+    QName = amqqueue:get_name(Amqqueue),
+
+    %% configure access to queue required for queue.declare
+    ok = check_resource_access(User, QName, configure, AuthzCtx),
+
+    case rabbit_vhost_limit:is_over_queue_limit(VHost) of
+        false ->
+            rabbit_core_metrics:queue_declared(QName),
+
+            case rabbit_queue_type:declare(Amqqueue, node()) of
+                {new, Q} when ?is_amqqueue(Q) ->
+                        rabbit_core_metrics:queue_created(QName),
+                    {ok, Q};
+                {existing, Q} when ?is_amqqueue(Q) ->
+                    rabbit_core_metrics:queue_created(QName),
+                    {ok, Q};
+                Other ->
+                    log_error(rabbit_misc:format("Failed to declare ~s: ~p", [rabbit_misc:rs(QName)]), Other, none),
+                    {error, queue_declare}
+            end;
+        {true, Limit} ->
+            log_error(rabbit_misc:format("cannot declare ~s because ", [rabbit_misc:rs(QName)]),
+                      rabbit_misc:format("queue limit ~p in vhost '~s' is reached",  [Limit, VHost]),
+                          none),
+            {error, queue_limit_exceeded}
+    end.
+
+routing_init_state() -> sets:new().
```

Comment:
use v2

Reply:
what is v2?

Reply:
https://www.erlang.org/doc/man/sets.html#description
> Erlang/OTP 24.0 introduced a new internal representation for sets which is more performant. Developers can use this new representation by passing the {version, 2} flag to [new/1](https://www.erlang.org/doc/man/sets.html#new-1) and [from_list/2](https://www.erlang.org/doc/man/sets.html#from_list-2), such as sets:new([{version, 2}]). This new representation will become the default in future Erlang/OTP versions. Functions that work on two sets, such as [union/2](https://www.erlang.org/doc/man/sets.html#union-2) and similar, will work with sets of different versions. In such cases, there is no guarantee about the version of the returned set. Explicit conversion from the old version to the new one can be done with sets:from_list(sets:to_list(Old), [{version,2}]).